src/app/admin/collectionCentresTable/CollectionCentresTable.tsx

```diff
@@ -0,0 +1,109 @@
+"use client";
+
+import React, { useState } from "react";
+import Table, { TableHeaders } from "@/components/Tables/Table";
+import styled from "styled-components";
+import Modal from "@/components/Modal/Modal";
+import Button from "@mui/material/Button";
+import DeleteIcon from "@mui/icons-material/Delete";
+import RefreshPageButton from "@/app/admin/RefreshPageButton";
+import { Schema } from "@/database_utils";
+import supabase from "@/supabaseClient";
+
+const DangerDialog = styled(Modal)`
+    & #deleteCollectionCentreDialog {
+        background-color: ${(props) => props.theme.error};
+    }
+`;
+
+export const OptionButtonDiv = styled.div`
+    display: flex;
+    padding-top: 1rem;
+    gap: 1rem;
+    justify-content: center;
+`;
+
+const collectionCentresTableHeaderKeysAndLabels: TableHeaders = [
+    ["name", "Name"],
+    ["acronym", "Acronym"],
+];
+
+interface Props {
+    collectionCentreData: Schema["collection_centres"][];
+}
+
+const CollectionCentresTables: React.FC<Props> = (props) => {
+    const [collectionCentreToDelete, setCollectionCentreToDelete] =
+        useState<Schema["collection_centres"]>();
+    const [refreshRequired, setRefreshRequired] = useState(false);
+
+    const collectionCentreOnDelete = (rowIndex: number): void => {
+        setCollectionCentreToDelete(props.collectionCentreData[rowIndex]); // TODO VFB-23 Change onDelete in table to return row
+    };
+
+    const onCollectionCentreDeleteConfirmation = async (): Promise<void> => {
+        const { error } = await supabase
+            .from("collection_centres")
+            .delete()
+            .eq("name", collectionCentreToDelete!.name);
+
+        // TODO VFB-23 Handle error on request to delete collection centre
+
+        if (error) {
+            throw new Error("Error deleting collection centre");
+        }
+
+        setCollectionCentreToDelete(undefined);
+        setRefreshRequired(true);
+    };
+
+    const onCollectionCentreDeleteCancellation = (): void => {
+        setCollectionCentreToDelete(undefined);
+    };
+
+    return (
+        <>
+            <Table
+                data={props.collectionCentreData}
+                headerKeysAndLabels={collectionCentresTableHeaderKeysAndLabels}
+                onDelete={collectionCentreOnDelete}
+                headerFilters={["name"]}
+            />
+
+            {refreshRequired ? (
+                <OptionButtonDiv>
+                    <RefreshPageButton />
+                </OptionButtonDiv>
+            ) : (
+                <></>
+            )}
+
+            <DangerDialog
+                header="DELETE COLLECTION CENTRE"
+                headerId="deleteCollectionCentreDialog"
+                isOpen={collectionCentreToDelete !== undefined}
+                onClose={onCollectionCentreDeleteCancellation}
+            >
+                Are you sure you want to delete collection centre{" "}
```

Comment:
What is that on the end for?

Reply:
You mean {" "}? It is to ensure there is a space after the word centre (even if you just put a space before the jsx tag it won't work and you need this syntax to force a space there

Reply:
Anyways Paopao pointed out a mistake here so it is no longer applicable